server/controllers/imageUpload.js

```diff
@@ -0,0 +1,19 @@
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+const path = require('path');
+const cloudinary = require('../utils/cloudinary')
+
+// @route POST /images
+// @desc Upload images to cloudanary
+// @access Private
+exports.uploadImages = asyncHandler(async (req, res, next) => {
+    // console.log(req.files);
+    const imageData = [];
+    for (const file of req.files) {
+        const result = await cloudinary.uploader.upload(file.path);
```

Comment:
as a suggestion, i wonder if it would be faster if you implement async calls to cloudlinary. from the postman video, it looks like uploading 3 images may take a good ~5 seconds. If you push all your await calls to a Promise.all https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all you can maybe send out all the calls at once and only when you retrieve all the info you can save them to `imageData` array

Reply:
@marouane-erouhi let me know if you need help with Promise.all ill be happy to assist you